serde/src/ser/mod.rs

```diff
@@ -607,6 +608,39 @@ pub trait Serializer {
         variant: &'static str,
         len: usize,
     ) -> Result<Self::SerializeStructVariant, Self::Error>;
+
+    /// Collect an iterator as a sequence.
+    ///
+    /// The default implementation serializes each item yielded by the iterator
+    /// using `Self::SerializeSeq`. Implementors should not need to override
+    /// this method.
+    fn collect_seq<I>(self, iter: I) -> Result<Self::Ok, Self::Error>
```

Comment:
Ok, then I don't understand the second question, if we can just check that the two hints are equal and use that, why specialise for `ExactSizeIterator` at all?